AzureMonitorAgent/agent.py

```diff
@@ -312,6 +311,27 @@ def install():
         return 0, ""
     elif protected_settings is None or len(protected_settings) == 0:
         default_configs["ENABLE_MCS"] = "true"
+
+        # Determine Managed Identity (MI) settings
+        # Nomenclature: Managed System Identity (MSI), System-Assigned Identity (SAI), User-Assigned Identity (UAI)
+        # Unspecified MI scenario: MSI returns SAI token if exists, otherwise returns UAI token if exactly one UAI exists, otherwise failure
+        # Specified MI scenario: MSI returns token for specified MI
+        if public_settings is not None and "authentication" in public_settings and "managedIdentity" in public_settings.get("authentication"):
+            managedIdentity = public_settings.get("authentication").get("managedIdentity")
+
+            if "identifier-name" not in managedIdentity or "identifier-value" not in managedIdentity:
+                log_and_exit("install", MissingorInvalidParameterErrorCode, 'Parameters "identifier-name" and "identifier-value" are both required in authentication.managedIdentity public setting')
+
+            identifier_name = managedIdentity.get("identifier-name")
+            identifier_value = managedIdentity.get("identifier-value")
+
+            if identifier_name not in ["object_id", "client_id", "mi_res_id"]:
+                log_and_exit("install", MissingorInvalidParameterErrorCode, 'Invalid identifier-name provided; must be "object_id", "client_id", or "mi_res_id"')
+
+            if not identifier_value:
+                log_and_exit("install", MissingorInvalidParameterErrorCode, 'Invalid identifier-value provided; cannot be empty')
+
+            default_configs["MANAGED_IDENTITY"] = "{0}#{1}".format(identifier_name, identifier_value)
```

Comment:
the first 2 are GUID-type strings; the latter is the resource of the managed identity. so none of these should have `#`, but customer could in theory pass whatever value they want here. McsManager will take everything after `object_id#`, `client_id#`, or `mi_res_id#`, so there isn't any issue such as with logic splitting on `#`. The token call will fail, but I don't know if it is worth validating GUID/resource ID format here.

Reply:
added GUID validation

Reply:
A bit of overkill for Contains('#') validation in object_id to avoid clashes with our special charater selection... :)
Maybe, we wanted to have GUID validation anyway as an input validation check for incoming parameters from security standpoint (e.g. to disallow limitless inputs that make break an agent / cause overflows), so sounds like a good choice if we are certain that incoming IDs are only GUID-like.